### ISSUE TITLE
Getting "Trouble accessing camera. I2C bus failure. Is camera plugged in?" in OctoPi Interface

### ISSUE THREAD

**Issue by jcalcote** [moof-src/ArduCamFocus#16]
Hello. I downloaded the ArduCamFocus OctoPi plugin as provided in the instructions I received with my "ArduCam Motorized Focus Camera on OctoPi" (SMP SKU: B0176). I followed the instructions carefully, adding the two dtparam options to the /boot/config.txt file and restarting the Pi. I also enabled i2c in raspi-config and restarted the Pi.

The hardware is a Raspberry Pi 4B. Here's my config: 
```
Linux aquilax2 6.1.21-v7l+ #1642 SMP Mon Apr  3 17:22:30 BST 2023 armv7l
...
------------------------------------------------------------------------------
OctoPrint version : 1.9.2
OctoPi version    : 1.0.0
------------------------------------------------------------------------------
```

The ArduCamFocus plugin seems to work as it actually does adjust the focus on the camera. But each time I click the left/right focus buttons in the UI, the UI changes and displays the message in the ticket description above:
```
Trouble accessing camera. I2C bus failure. Is camera plugged in?
```
However, when I refresh the browser, the UI returns to the original display as if there was no error -- AND -- displays the updated focus value. 

For instance, when I initially start the OctoPrint UI and change to the "control" tab, I can see the ArduCamFocus plugin enhancement on the lower right under the General column and to the right of the Flow Rate Modifier - as you'd expect. I see the current focus value as 100. When I click the right (increase) button, the screen immediately changes so that the ArduCamFocus plugin UI extensions move to the buttom left of the display, with the above message under the ArduCamFocus title, and the left/right buttons (disabled) under that line. When I refresh the browser, the display returns to the original display first described - BUT - the focus value has changed to 110.

Also, the "@ARDUCAMFOCUSSET <value>" gcode command works as well - if I enter a value of 200, for instance, it does change the focus to that value, but when I send the gcode (even if I send it from Cura's octoprint plugin), the OctoPrint Control interface changes to the ArduCamFocus plugin error screen (as described above). As you'd expect, a browser refresh fixes the display problem and the focus value has been updated to 200.

In summary, the plugin seems to work, but requires a UI refresh between modifications of the focus value. This feels like some kind of timing issue - perhaps the plugin isn't waiting long enough for the i2c bus to stabilize before trying to read the camera's focus value?

**Comment by moof-src** [comment]
I haven’t been maintaining the plugin.  I haven’t used my printer for a
couple of years.  I have two guesses: 1) my plugin is too old and now
incompatible. 2) the cable is not seated well in the connector and is
getting infrequent communication errors.  I don’t have a good answer.

**Comment by jcalcote** [comment]
Well, thanks for the quick feedback anyway. 

Listen, I know this isn't your responsibility (as a third-party contributor), but strangely my camera came with printed instructions for installing your plugin, so would you happen to know if ArduCam is providing an updated plugin or driver for their camera? 

Thanks, regardless.
John

**Comment by moof-src** [comment]
I don’t know.  I hope they would have done that. Anybody seeing this
thread, please reply with info on newer plugins.  Thanks.
